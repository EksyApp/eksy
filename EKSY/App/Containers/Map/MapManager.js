import React from 'react'
import * as Actions from '../../Actions'
import Store from '../../Store'
import BackgroundGeolocation from 'react-native-mauron85-background-geolocation'

let instance = null
let idCounter = 0

class MapManager {

	constructor() {
		if (!instance) {
			this._map = null
			this._currentLocationMoveRequested = false
			this._reduxState = null
			this.initStore()
			// this.store.subscribe(() => this._storeListener())
			this.startLocationWatcher()
			instance = this
		}
		return instance
	}

	async initStore() {
		this.store = await Store()
		this.store.subscribe(() => this.storeListener())
	}

	static getNextID() {
		idCounter++
		return idCounter
	}

	storeListener() {
		this._reduxState = this.store.getState()
		this.handleFlyingToCurrentLocation()
	}

	handleFlyingToCurrentLocation() {
		if (this._currentLocationMoveRequested) {
			this.goToCurrentPosition()
		}
	}

	startLocationWatcher() {
		BackgroundGeolocation.configure({
			desiredAccuracy: 10,
      stationaryRadius: 10,
      distanceFilter: 10,
      locationTimeout: 30,
      debug: false,
      startOnBoot: false,
      stopOnTerminate: true,
      locationProvider: BackgroundGeolocation.provider.ANDROID_ACTIVITY_PROVIDER,
      interval: 10000,
      fastestInterval: 5000,
      activitiesInterval: 10000,
      stopOnStillActivity: true
		})

		BackgroundGeolocation.on('location', (location) => {
				this.store.dispatch(Actions.updateLocation(location))
				this.store.dispatch(Actions.locationKnown(true))
				this._map.forceUpdate()
		})

		BackgroundGeolocation.on('stationary', () => {})

		BackgroundGeolocation.on('error', (error) => {
			console.log('Geolocation error: ' + error)
			this.store.dispatch(Actions.locationKnown(false))
		})

		BackgroundGeolocation.start(() => {
			console.log('BackgroundGeolocation started')
		})
		// this.watchID = navigator.geolocation.watchPosition(
		// 		(position) => {
		// 			this.store.dispatch(Actions.updateLocation(position.coords))
		// 			this.store.dispatch(Actions.locationKnown(true))
		// 			this._map.forceUpdate()
		// 		},
		// 		(error) => this.store.dispatch(Actions.locationKnown(false)),
		// 		{enableHighAccuracy: false, timeout: 500, maximumAge: 0, distanceFilter: 3}
		// )
	}

	setMapObject(map) {
		this._map = map
	}

	goToCurrentPosition() {
		if (this._reduxState && this._reduxState.map.location.isKnown) {
			this._currentLocationMoveRequested = false
			this.flyToPosition(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
		} else {
			this._currentLocationMoveRequested = true
		}
	}

	flyToPosition(latitude, longitude) {
		let position = {
			latitude: latitude,
			longitude: longitude
		}
		this._map.animateToCoordinate(position, 1000)
	}

}

export default MapManager
