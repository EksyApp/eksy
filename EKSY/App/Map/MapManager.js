import React from 'react'
import * as Actions from '../Actions'
import configureStore from '../Store'
import Marker from './Marker'

let instance = null
let idCounter = 0

class MapManager {

  constructor () {
    if (!instance) {
      this._markers = new Map()
      this._map = null
      this._currentLocationMoveRequested = false
      this._reduxState = null
      this.initStore()
      // this.store.subscribe(() => this.storeListener())
      this.startLocationWatcher()
      instance = this
    }
    return instance
  }

  async initStore () {
    this.store = await configureStore()
    this.store.subscribe(() => this.storeListener())
  }

  static getNextID () {
    idCounter++
    return idCounter
  }

  storeListener () {
    this._reduxState = this.store.getState()
    this.handleFlyingToCurrentLocation()
  }

  handleFlyingToCurrentLocation () {
    if (this._currentLocationMoveRequested) {
      this.goToCurrentPosition()
    }
  }

  startLocationWatcher () {
    navigator.geolocation.watchPosition(
				(position) => {
  this.store.dispatch(Actions.updateLocation(position.coords))
  this.store.dispatch(Actions.locationKnown(true))
  this._map.forceUpdate()
},
				(error) => this.store.dispatch(Actions.locationKnown(false)),
				{enableHighAccuracy: false, timeout: 500, maximumAge: 500, distanceFilter: 10}
		)
  }

  addMarker (key, marker) {
    let markerComponent = <Marker title={marker.title} images={marker.images} latitude={marker.latitude} longitude={marker.longitude} text={marker.text} key={MapManager.getNextID()} />
    this._markers.set(key, markerComponent)
    this._map.forceUpdate()
  }

  removeMarker (key) {
    this._markers.delete(key)
    this._map.forceUpdate()
  }

  getMarkers () {
    return this._markers
  }

  setMapObject (map) {
    this._map = map
  }

  goToCurrentPosition () {
    if (this._reduxState && this._reduxState.map.location.isKnown) {
      this._currentLocationMoveRequested = false
      this.flyToPosition(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
    } else {
      this._currentLocationMoveRequested = true
    }
  }

  flyToPosition (latitude, longitude) {
    let position = {
      latitude: latitude,
      longitude: longitude
    }
    this._map.animateToCoordinate(position, 1000)
  }

}

export default MapManager
