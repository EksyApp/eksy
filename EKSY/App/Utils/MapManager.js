import React from 'react'
import * as Actions from '../Actions/index'
import Store from '../Store/index'
import Dao from "../Dao/Dao";
import {PermissionsAndroid} from 'react-native'
//import BackgroundGeolocation from 'react-native-mauron85-background-geolocation'

let instance = null

// used to give commands to the Google Map
class MapManager {

	constructor() {
		if (!instance) {
			this._map = null
			this._reduxState = null
			this.dao = new Dao()
			this.initStore()
			instance = this
		}
		return instance
	}

	async initStore() {
		this.store = await Store()
		this.store.subscribe(() => this.storeListener())
		this.startLocationWatcher()
	}

	storeListener() {
		this._reduxState = this.store.getState()
	}

	async startLocationWatcher() {
		this._reduxState = this.store.getState()
		
		if(this._reduxState.map.location.isKnown && this._reduxState.map.location.latitude != null) {
			this.dao.updateLocation(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
		}
		
		try {
			const granted = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION)
			if (granted === PermissionsAndroid.RESULTS.GRANTED) {
				this.watchID = navigator.geolocation.watchPosition(
						(position) => {
							this.dao.updateLocation(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
							this.store.dispatch(Actions.updateLocation(position.coords))
							this.store.dispatch(Actions.locationKnown(true))
							this._map.forceUpdate()
						},
						(error) => this.store.dispatch(Actions.locationKnown(false)),
						{enableHighAccuracy: false, timeout: 500, maximumAge: 0, distanceFilter: 3}
				)
			} else {
				this.store.dispatch(Actions.locationKnown(false))
			}
		} catch (err) {
			this.store.dispatch(Actions.locationKnown(false))
		}
		
		
	}

	setMapObject(map) {
		this._map = map
	}

	goToCurrentPosition() {
		this.flyToPosition(this._reduxState.map.location.latitude, this._reduxState.map.location.longitude)
	}

	flyToPosition(latitude, longitude) {
		let position = {
			latitude: latitude,
			longitude: longitude
		}
		this._map.animateToCoordinate(position, 1000)
	}

}

export default MapManager
