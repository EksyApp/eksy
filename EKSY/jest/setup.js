jest.mock('Linking', () => {
	return {
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
		openURL: jest.fn(),
		canOpenURL: jest.fn(),
		getInitialURL: jest.fn(),
		requestPermissions: jest.fn(),
		configure: jest.fn()
	}
})

jest.mock('react-native-maps', () => {
	const React = require.requireActual('react');
	const MapView = require.requireActual('react-native-maps');
	
	class MockCallout extends React.Component {
		render() {
			return React.createElement('Callout', this.props, this.props.children);
		}
	}
	
	class MockMarker extends React.Component {
		render() {
			return React.createElement('Marker', this.props, this.props.children);
		}
	}
	
	class MockMapView extends React.Component {
		render() {
			return React.createElement('MapView', this.props, this.props.children);
		}
	}
	
	class MockMapViewAnimated extends React.Component {
		render() {
			return React.createElement('MapView', this.props, this.props.children);
		}
	}
	
	class MockPolyline extends React.Component {
		render() {
			return React.createElement('Polyline', this.props, this.props.children);
		}
	}
	
	class MockCircle extends React.Component {
		render() {
			return React.createElement('Circle', this.props, this.props.children);
		}
	}
	
	MockCallout.propTypes = MapView.Callout.propTypes;
	MockMarker.propTypes = MapView.Marker.propTypes;
	MockMapView.propTypes = MapView.propTypes;
	MockMapView.Marker = MockMarker;
	MockMapView.Callout = MockCallout;
	MockMapView.Animated = MockMapViewAnimated;
	MockMapView.Polyline = MockPolyline;
	MockMapView.Circle = MockCircle;
	
	return MockMapView;
});


jest.mock('react-native-fetch-blob', () => {
	return {
		DocumentDir: () => {},
		fetch: () => {},
		base64: () => {},
		android: () => {},
		ios: () => {},
		config: () => {},
		session: () => {},
		fs: {
			dirs: {
				MainBundleDir: () => {},
				CacheDir: () => {},
				DocumentDir: () => {},
			},
		},
		wrap: () => {},
		polyfill: {
			Fetch: class Fetch {
				constructor(params) {
				
				}
				
				build = jest.fn()
			}
		},
		JSONStream: () => {}
	}
})

jest.mock('firebase', () => {
	
	const update = jest.fn(() => {
		return Promise.resolve()
	})
	
	const ref = jest.fn(() => {
		return {
			child: jest.fn(() => {
				return ref
			}),
			update: update
		}
	})
	
	return {
		initializeApp: jest.fn(),
		database: jest.fn(() =>{
			return {
				ref:ref
			}
		})
	}
})

jest.mock('react-native-snap-carousel', () => {
	const React = require.requireActual('react');
	const {View} = require('react-native')
	
	class MockCarousel extends React.Component {
		
		render() {
			return (<View {...this.props} />);
		}
	}
	
	MockCarousel.style = {}
	return MockCarousel
})