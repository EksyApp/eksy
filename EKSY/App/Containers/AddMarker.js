import React, {Component} from 'react'
import {View, Text, TextInput, StyleSheet} from 'react-native'
// import {Container, Content, Header, Body, Title, Left, Button} from 'native-base'
import PointSelector from '../Components/PointSelector'
import Marker from '../Map/Marker'
import MapManager from '../Map/MapManager'
import Header from '../Components/Header'
import * as ReduxActions from '../Actions'
import {Actions} from 'react-native-router-flux'
import {connect} from 'react-redux'
import {backgroundColor, frontgroundColor} from '../Theme'
import {Grid, Row, FormInput, Button} from 'react-native-elements'
import ActionButton from '../Components/ActionButton'

class AddMarker extends Component {
	
	constructor(props) {
		super(props)
		
		this.state = {
			text: ""
		}
		
		this.mapManager = new MapManager();
	}
	
	addNewMarker() {
		let latitude = this.props.currentRegion.latitude;
		let longitude = this.props.currentRegion.longitude;
		let marker = <Marker latitude={latitude} longitude={longitude} text={this.state.text}/>;
		this.mapManager.addMarker(marker);
		setTimeout(() => this.mapManager.flyToPosition(latitude, longitude), 1000)
		Actions.mapView();
		// this.mapManager.update();
	}
	
	
	render() {
		return (
				// <Container>
				//   <Header>
				//
				//     <MenuButton />
				//     <Left />
				//     <Body>
				//       <Title>Add a new marker</Title>
				//     </Body>
				//   </Header>
				//   <Content style={Style.container}>
				//     <PointSelector onChange={(region) => this.state.region = region} style={Style.map} />
				//     <TextInput
				//       onChange={(event) => this.state.text = event.nativeEvent.text}
				//       multiline
				//       numberOfLines={10}
				//       style = {Style.textInput}
				//     />
				//     <Button block onPress={() => this.addNewMarker()} style={Style.button}>
				//       <Text style={Style.text}>Add</Text>
				//     </Button>
				//   </Content>
				//
				// </Container>
				
				<View style={styles.container}>
					<Header title="Add Marker" menuButtonPress={this.props.menuButtonPress}/>
					<Grid>
						<Row size={2} containerStyle={styles.row}>
							<PointSelector onChange={(region) => {
								this.props.regionChange(region)
							}} currentRegion={this.props.currentRegion} style={styles.map}/>
						</Row>
						<Row size={2} containerStyle={styles.row}>
							<FormInput onChangeText={(text) => this.setState({text: text})} multiline numberOfLines={8} style={styles.textArea}/>
						</Row>
						<Row size={1} containerStyle={styles.row}>
							<ActionButton title="Add" onPress={() => {
								this.addNewMarker()
							}}/>
						</Row>
					</Grid>
				</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: backgroundColor,
		flex: 1
	},
	
	row: {
		width: '100%'
	},
	
	map: {
		flex: 1,
	},
	
	textArea: {
		flex: 1,
		textAlignVertical: 'top',
		width: '100%',
		backgroundColor: frontgroundColor,
		marginTop: 10
	}
	
	
})

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		menuButtonPress: () => {
			dispatch(ReduxActions.drawerOpen())
		},
		regionChange: (region) => {
			dispatch(ReduxActions.updateRegion(region))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(AddMarker)
