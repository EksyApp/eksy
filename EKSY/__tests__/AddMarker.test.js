import React from 'react'
import '../_mocks_/setup'
import {shallow} from 'enzyme'
import {AddMarker} from "../App/Containers/AddMarker";
import MapManager from "../App/Containers/Map/MapManager"
import renderer from 'react-test-renderer'

MapManager.prototype.startLocationWatcher = jest.fn()
MapManager.prototype.storeListener = jest.fn()
MapManager.prototype.goToCurrentPosition = jest.fn()
MapManager.prototype.flyToPosition = jest.fn()

jest.mock('react-native-router-flux', () => ({
	Actions: {
		pop: jest.fn()
	}
}))

jest.mock('react-native-fetch-blob', () => {
  return {
    DocumentDir: () => {},
    polyfill: () => {},
		Fetch: () => {}
  }
});

let currentRegion = {
	latitude: 60.184356,
	longitude: 24.949326,
	latitudeDelta: 0.1,
	longitudeDelta: 0.1,
}

// let addMarker = shallow(<AddMarker currentRegion={currentRegion} />);
let mapManager = new MapManager();

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  return MockMapView;
});


describe("AddMarker", () => {

	this.addNewMarkerMock = jest.fn()

	this.addNewMarkerMock.bind(this)

	it('renders correctly', () => {
	  const tree = renderer.create(
	    <AddMarker currentRegion={currentRegion}/>
	  ).toJSON();
	  expect(tree).toMatchSnapshot();
	});

	it('Adds a marker without content', () => {
		let addMarker = shallow(<AddMarker currentRegion={currentRegion} addNewMarker={this.addNewMarkerMock} />);
		addMarker.find('Button').last().simulate('press')
		expect(this.addNewMarkerMock).toBeCalledWith({
			latitude: currentRegion.latitude,
			longitude: currentRegion.longitude,
			text: '',
			title: '',
			images: []
		})
	})

	it('Adds a marker with title', () => {
		let addMarker = shallow(<AddMarker currentRegion={currentRegion} addNewMarker={this.addNewMarkerMock} />);
		addMarker.find('[label="Title"]').simulate('changeText', "A Nice Title")
		addMarker.find('Button').last().simulate('press')
		expect(this.addNewMarkerMock).toBeCalledWith({
			latitude: currentRegion.latitude,
			longitude: currentRegion.longitude,
			text: '',
			title: 'A Nice Title',
			images: []
		})
	})

	it('Adds a marker with text', () => {
		let addMarker = shallow(<AddMarker currentRegion={currentRegion} addNewMarker={this.addNewMarkerMock} />);
		addMarker.find('[label="Text"]').simulate('changeText', "Some great text")
		addMarker.find('Button').last().simulate('press')
		expect(this.addNewMarkerMock).toBeCalledWith({
			latitude: currentRegion.latitude,
			longitude: currentRegion.longitude,
			text: 'Some great text',
			title: '',
			images: []
		})
	})

	it('Adds a marker with a title and text', () => {
		let addMarker = shallow(<AddMarker currentRegion={currentRegion} addNewMarker={this.addNewMarkerMock} />);
		addMarker.find('[label="Title"]').simulate('changeText', "A Nice Title")
		addMarker.find('[label="Text"]').simulate('changeText', "Some great text")
		addMarker.find('Button').last().simulate('press')
		expect(this.addNewMarkerMock).toBeCalledWith({
			latitude: currentRegion.latitude,
			longitude: currentRegion.longitude,
			text: 'Some great text',
			title: 'A Nice Title',
			images: []
		})
	})

	// it('Adds a marker with a valid FastImage', () => {
	// 	let addMarker = shallow(<AddMarker currentRegion={currentRegion} addNewMarker={this.addNewMarkerMock} />);
	// 	addMarker.find('[label="FastImage URL"]').simulate('changeText', "http://static.wixstatic.com/media/88e4c2_dde3ecf82909493f94bb32a60fe1a8c6~mv2.jpg")
	// 	addMarker.find('Button').first().simulate('press')
	// 	return new Promise(resolve => setTimeout(resolve, 20)).then(() => {
	// 		addMarker.find('Button').last().simulate('press')
	// 		expect(this.addNewMarkerMock).toBeCalledWith({
	// 			latitude: currentRegion.latitude,
	// 			longitude: currentRegion.longitude,
	// 			text: '',
	// 			title: '',
	// 			images: ["http://static.wixstatic.com/media/88e4c2_dde3ecf82909493f94bb32a60fe1a8c6~mv2.jpg"]
	// 		})
	// 	})
	// })

	it('imageurlerror sets state correctly', () => {
		let addMarker = shallow(<AddMarker currentRegion={currentRegion} addNewMarker={this.addNewMarkerMock} />);
		expect(addMarker.instance().state.imageResponse).toBe("")
		addMarker.instance()._imageUrlError()
		expect(addMarker.instance().state.imageResponse).toBe("URL not valid")
	})

	//it('Doesnt add an image when URL is invalid', () => {
	//	return new Promise(resolve => setTimeout(resolve, 50)).then(() => {
	//		let addMarker = shallow(<AddMarker currentRegion={currentRegion}/>);
	//		addMarker.find('[label="FastImage URL"]').simulate('changeText', "http://www.google.com")
	//		console.log(addMarker.instance().state)
	//		addMarker.find('Button').first().simulate('press')
	//		return new Promise(resolve => setTimeout(resolve, 20)).then(() => {
	//			addMarker.find('Button').last().simulate('press')
	//			//expect(mapManager.getMarkers()[5].props.images.length).toBe(0)
	//			expect(mapManager.getMarkers()[5].props.images[0].width).toBe("http://static.wixstatic.com/media/88e4c2_dde3ecf82909493f94bb32a60fe1a8c6~mv2.jpg")
	//		})
	//	})
	//})

})
