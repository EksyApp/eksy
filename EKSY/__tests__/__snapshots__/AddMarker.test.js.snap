// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMarker renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(243,244,229)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(226,227,217)",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "height": 50,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 50,
              "zIndex": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 35,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgb(101,114,122)",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î—„
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(101,114,122)",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Add Marker
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(243,244,229)",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": "50%",
          "width": "100%",
        }
      }
    >
      <MapView
        initialRegion={
          Object {
            "latitude": 60.184356,
            "latitudeDelta": 0.1,
            "longitude": 24.949326,
            "longitudeDelta": 0.1,
          }
        }
        onRegionChange={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Marker
          coordinate={
            Object {
              "latitude": 60.184356,
              "longitude": 24.949326,
            }
          }
        />
      </MapView>
    </View>
    <RCTScrollView>
      <View>
        <View
          style={undefined}
        >
          <View>
            <View
              style={
                Array [
                  Object {},
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#86939e",
                        "fontSize": 15,
                        "fontWeight": "bold",
                        "marginBottom": 1,
                        "marginLeft": 20,
                        "marginRight": 20,
                        "marginTop": 15,
                      },
                      Object {
                        "color": "rgb(101,114,122)",
                        "fontSize": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Title
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#bdc6cf",
                    "borderBottomWidth": 1,
                    "marginLeft": 20,
                    "marginRight": 20,
                  },
                  Object {
                    "backgroundColor": "rgb(226,227,217)",
                    "margin": 10,
                  },
                ]
              }
            >
              <TextInput
                autoCorrect={false}
                onChangeText={[Function]}
                placeholder={undefined}
                secureTextEntry={undefined}
                selectionColor="#86939e"
                style={
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 17.5,
                      "height": 36,
                      "width": 750,
                    },
                    Object {
                      "color": "rgb(89,85,78)",
                      "fontSize": 18,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value={undefined}
              />
            </View>
          </View>
          <View>
            <View
              style={
                Array [
                  Object {},
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#86939e",
                        "fontSize": 15,
                        "fontWeight": "bold",
                        "marginBottom": 1,
                        "marginLeft": 20,
                        "marginRight": 20,
                        "marginTop": 15,
                      },
                      Object {
                        "color": "rgb(101,114,122)",
                        "fontSize": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Text
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#bdc6cf",
                    "borderBottomWidth": 1,
                    "marginLeft": 20,
                    "marginRight": 20,
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(226,227,217)",
                      "margin": 10,
                    },
                  ],
                ]
              }
            >
              <TextInput
                autoCorrect={undefined}
                multiline={true}
                onChange={[Function]}
                onChangeText={[Function]}
                placeholder={undefined}
                secureTextEntry={undefined}
                selectionColor="#86939e"
                style={
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 17.5,
                      "height": 36,
                      "width": 750,
                    },
                    Array [
                      Object {
                        "color": "rgb(89,85,78)",
                        "fontSize": 18,
                      },
                      Object {
                        "height": 45,
                      },
                    ],
                  ]
                }
                underlineColorAndroid="transparent"
                value={undefined}
              />
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(101,114,122)",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "rgb(230,232,227)",
                "borderColor": "rgb(101,114,122)",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 5,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "rgb(101,114,122)",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              Add image...
            </Text>
          </View>
          <RCTScrollView
            dataSource={
              ListViewDataSource {
                "items": 2,
              }
            }
            renderRow={[Function]}
            renderScrollComponent={[Function]}
            style={undefined}
          >
            <View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "rgb(226,227,217)",
                    "borderColor": "#ededed",
                    "borderRadius": 3,
                    "borderWidth": 1,
                    "margin": 5,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "opacity": 1,
                    "padding": 10,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                      false,
                      false,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    onLongPress={undefined}
                    onPress={undefined}
                    style={
                      Array [
                        Object {
                          "color": "#bfbfbf",
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    ï‚–
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontWeight": "bold",
                            "marginLeft": 10,
                            "marginRight": 10,
                          },
                          Array [
                            Object {
                              "color": "rgb(89,85,78)",
                            },
                            undefined,
                          ],
                          undefined,
                        ],
                      ]
                    }
                  >
                    Has violent content
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "rgb(226,227,217)",
                    "borderColor": "#ededed",
                    "borderRadius": 3,
                    "borderWidth": 1,
                    "margin": 5,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "opacity": 1,
                    "padding": 10,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                      false,
                      false,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    onLongPress={undefined}
                    onPress={undefined}
                    style={
                      Array [
                        Object {
                          "color": "#bfbfbf",
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    ï‚–
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "#43484d",
                            "fontWeight": "bold",
                            "marginLeft": 10,
                            "marginRight": 10,
                          },
                          Array [
                            Object {
                              "color": "rgb(89,85,78)",
                            },
                            undefined,
                          ],
                          undefined,
                        ],
                      ]
                    }
                  >
                    Has non-accessible content
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 10,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "rgb(230,232,227)",
                "borderColor": "rgb(101,114,122)",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 5,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "rgb(101,114,122)",
                  "fontSize": 16,
                  "fontWeight": "600",
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              Add marker
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
